utilities/io_tools/output_tools.py

```diff
@@ -75,9 +74,27 @@ def write_analysis_output(results, outfile, args, update_name=True):
             os.makedirs(args.outfolder)
         outfile = os.path.join(args.outfolder, outfile)
 
+    if args.appendOutputFile:
+        outfile = args.appendOutputFile
+        if os.path.isfile(outfile):
+            logger.info(f"Analysis output will be appended to file {outfile}")
+            open_as="a"
+        else:
+            logger.warning(f"Analysis output requested to be appended to file {outfile}, but the file does not exist yet, it will be created instead")
+            open_as="w"
+    else:
+        if os.path.isfile(outfile):
+            logger.warning(f"Output file {outfile} exists already, it will be overwritten")
+        open_as="w"
+
     time0 = time.time()
-    with h5py.File(outfile, 'w') as f:
-        narf.ioutils.pickle_dump_h5py("results", results, f)
+    with h5py.File(outfile, open_as) as f:
+        for k, v in results.items():
+            logger.debug(f"Pickle and dump {k}")
+            narf.ioutils.pickle_dump_h5py(k, v, f)
+
+        if "meta_info" not in f.keys():
```

Comment:
In case you run multiple times to analyse different processes, do I understand correctly that you only get the first original command in the metadata ? It's probably ok, since the only difference should be a call to the option to filter some processes, but do we have/want to keep track of all the calls somehow ?

Reply:
Yes, only the metadata of the first call is stored. We could change this, for example adding metadata of the additional calls with other keys. I didn't implement this to keep the scope of this feature small, since I consider the feature to be an "expert" option and would not suggest using it in general. There are downsides running the datasets separately and appending the output files gradually, for example, each time the helpers/graphs need to be initialized again. It's better to run all samples in one call if possible. So, at the moment I think this feature should only be used in certain cases, for example, if the memory is not enough (as it is the case for the unfolding). I suggest that we keep it like this for the moment and if it turns out that for some reason more people use the feature regularly we can support it better.